internal/oci/uvm.go

```diff
@@ -123,6 +123,19 @@ const (
 	annotationVPCIEnabled                = "io.microsoft.virtualmachine.lcow.vpcienabled"
 	annotationStorageQoSBandwidthMaximum = "io.microsoft.virtualmachine.storageqos.bandwidthmaximum"
 	annotationStorageQoSIopsMaximum      = "io.microsoft.virtualmachine.storageqos.iopsmaximum"
+	// SaveAsTemplate annotation must be used with a pod & container creation request.
+	// If this annotation is present in the request then it will save the UVM (pod)
+	// and the container(s) inside it as a template. However, this also means that this
+	// pod and the containers inside this pod will permananetly stay in the
+	// paused/templated state and can not be resumed again.
+	annotationSaveAsTemplate = "io.microsoft.virtualmachine.saveastemplate"
+	// This annotation should be used when creating a pod or a container from a template.
+	// When creating a pod from a template use the ID of the templated pod as the
+	// TemplateID and when creating a container use the ID of the templated container as
+	// the TemplateID. It is the client's responsibility to make sure that the sandbox
+	// within which a cloned container needs to be created must also be created from the
+	// same template.
```

Comment:
Do we have no way of ensuring this on our end? what happens if it is not?

Reply:
We will basically have to maintain some kind of mapping of existing templates and the pod ID and container ID in those templates and then for every clone request we will have to verify against those mappings. As of now if you pass the wrong templateID during pod creation the request will simply fail by saying templateID not found. If you pass a wrong templateID in the container creation request then it will fail with container not found error. I think these errors are fine for now considering the extra work that will be required for maintain these mappings.